modules/stochastic_tools/src/samplers/SobolSampler.C

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "SobolSampler.h"
+
+template <>
+InputParameters
+validParams<SobolSampler>()
+{
+  InputParameters params = validParams<Sampler>();
+  params.addClassDescription("Sobol variance-based sensitivity analysis Sampler.");
+  params.addRequiredParam<unsigned int>(
+      "n_samples", "Number of Monte Carlo samples to perform for each distribution.");
+  params.addParam<std::vector<unsigned int>>("seeds",
+                                             std::vector<unsigned int>({19800624, 19801009}),
```

Comment:
I am no expert on this, but the problem with creating one from the other is that B is then always some offset of A, which seems like a bad idea.

This class should error if the seeds for A and B are the same.